nipst/nipst.go

```diff
@@ -197,18 +196,14 @@ func (nb *NIPSTBuilder) BuildNIPST(challenge *common.Hash) (*types.NIPST, error)
 	if nipst.PostProof == nil {
 		nb.log.Info("starting PoST execution (challenge: %x)", nb.state.PoetProofRef)
 
-		proof, err := nb.postProver.execute(nb.id, nb.state.PoetProofRef, defTimeout)
+		proof, err := nb.postProver.Execute(nb.state.PoetProofRef)
 		if err != nil {
 			return nil, fmt.Errorf("failed to execute PoST: %v", err)
 		}
 
-		res, err := nb.verifyPost(proof, nb.postCfg.SpacePerUnit, nb.postCfg.NumProvenLabels, nb.postCfg.Difficulty)
-		if err != nil {
+		if err := nb.verifyPost(proof, cfg.SpacePerUnit, cfg.NumProvenLabels, cfg.Difficulty); err != nil {
```

Comment:
Do you see any reason to keep this defensive behavior?

Reply:
Probably not at this point. Removed.